File: openbis_api/source/java/ch/ethz/sis/openbis/generic/shared/api/v3/dto/entity/history/HistoryEntry.java
/*
 * Copyright 2014 ETH Zuerich, CISD
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package ch.ethz.sis.openbis.generic.shared.api.v3.dto.entity.history;

import ch.ethz.sis.openbis.generic.shared.api.v3.dto.entity.person.Person;
import ch.ethz.sis.openbis.generic.shared.api.v3.dto.fetchoptions.history.HistoryEntryFetchOptions;
import ch.ethz.sis.openbis.generic.shared.api.v3.exceptions.NotFetchedException;
import ch.systemsx.cisd.base.annotation.JsonObject;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.io.Serializable;
import java.util.Date;

/**
 * Class automatically generated with {@link ch.ethz.sis.openbis.generic.shared.api.v3.dto.generators.DtoGenerator}
 */
@JsonObject("dto.entity.history.HistoryEntry")
public abstract class HistoryEntry implements Serializable
{
    private static final long serialVersionUID = 1L;

    @JsonProperty
    private HistoryEntryFetchOptions fetchOptions;

    @JsonProperty
    private Date validFrom;

    @JsonProperty
    private Date validTo;

    @JsonProperty
    private Person author;

    // Method automatically generated with {@link ch.ethz.sis.openbis.generic.shared.api.v3.dto.generators.DtoGenerator}
    @JsonIgnore
    public HistoryEntryFetchOptions getFetchOptions()
    {
        return fetchOptions;
    }

    // Method automatically generated with {@link ch.ethz.sis.openbis.generic.shared.api.v3.dto.generators.DtoGenerator}
    public void setFetchOptions(HistoryEntryFetchOptions fetchOptions)
    {
        this.fetchOptions = fetchOptions;
    }

    // Method automatically generated with {@link ch.ethz.sis.openbis.generic.shared.api.v3.dto.generators.DtoGenerator}
    @JsonIgnore
    public Date getValidFrom()
    {
        return validFrom;
    }

    // Method automatically generated with {@link ch.ethz.sis.openbis.generic.shared.api.v3.dto.generators.DtoGenerator}
    public void setValidFrom(Date validFrom)
    {
        this.validFrom = validFrom;
    }

    // Method automatically generated with {@link ch.ethz.sis.openbis.generic.shared.api.v3.dto.generators.DtoGenerator}
    @JsonIgnore
    public Date getValidTo()
    {
        return validTo;
    }

    // Method automatically generated with {@link ch.ethz.sis.openbis.generic.shared.api.v3.dto.generators.DtoGenerator}
    public void setValidTo(Date validTo)
    {
        this.validTo = validTo;
    }

    // Method automatically generated with {@link ch.ethz.sis.openbis.generic.shared.api.v3.dto.generators.DtoGenerator}
    @JsonIgnore
    public Person getAuthor()
    {
        if (getFetchOptions().hasAuthor())
        {
            return author;
        }
        else
        {
            throw new NotFetchedException("Author has not been fetched.");
        }
    }

    // Method automatically generated with {@link ch.ethz.sis.openbis.generic.shared.api.v3.dto.generators.DtoGenerator}
    public void setAuthor(Person author)
    {
        this.author = author;
    }

}

File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/dto/entity/history/HistoryEntry.js
define([ "stjs", "util/Exceptions" ], function(stjs, exceptions) {
	var HistoryEntry = function() {
	};
	stjs.extend(HistoryEntry, null, [], function(constructor, prototype) {
		prototype['@type'] = 'dto.entity.history.HistoryEntry';
		constructor.serialVersionUID = 1;
		prototype.fetchOptions = null;
		prototype.validFrom = null;
		prototype.validTo = null;
		prototype.author = null;

		prototype.getFetchOptions = function() {
			return this.fetchOptions;
		};
		prototype.setFetchOptions = function(fetchOptions) {
			this.fetchOptions = fetchOptions;
		};
		prototype.getValidFrom = function() {
			return this.validFrom;
		};
		prototype.setValidFrom = function(validFrom) {
			this.validFrom = validFrom;
		};
		prototype.getValidTo = function() {
			return this.validTo;
		};
		prototype.setValidTo = function(validTo) {
			this.validTo = validTo;
		};
		prototype.getAuthor = function() {
			if (this.getFetchOptions().hasAuthor()) {
				return this.author;
			} else {
				throw new exceptions.NotFetchedException("Author has not been fetched.");
			}
		};
		prototype.setAuthor = function(author) {
			this.author = author;
		};
	}, {
		fetchOptions : "HistoryEntryFetchOptions",
		validFrom : "Date",
		validTo : "Date",
		author : "Person"
	});
	return HistoryEntry;
})